service/service.go

```diff
@@ -38,43 +34,42 @@ func (s Status) String() string {
 	}
 }
 
-// OsSvc is OS service
-type OsSvc struct {
+// Svc is OS service
+type Svc struct {
 	// Name of the service
 	name string
-	// Service Init system
-	sysInit string
+	// System init type
+	sysInit SysInit
 }
 
-// NewOsSvc creates new Service or returns error if the requested service type is not supported
-func NewOsSvc(name, sysInitType string) (*OsSvc, error) {
-	sysInitTypes := []string{"upstart", "systemd", "sysv"}
-	supported := make(map[string]bool)
-	for _, sysInitType := range sysInitTypes {
-		supported[sysInitType] = true
+// NewSvc creates new Service or returns error if the requested service type is not supported
+func NewSvc(name, sysInitType string) (*Svc, error) {
+	sysInit, err := NewSysInit(sysInitType)
+	if err != nil {
+		return nil, err
 	}
 
-	if !supported[sysInitType] {
-		return nil, fmt.Errorf("Unsupported service init type: %s", sysInitType)
+	if name == "" {
```

Comment:
I wouldn't test this. The "constructor" is expecting a string, if you send an empty string you know what you are doing, or... you want to break something.